src/confdb/confdb.c

```diff
@@ -2116,6 +2117,15 @@ static int confdb_merge_parent_domain(const char *name,
         }
     }
 
+    ret = ldb_transaction_start(cdb->ldb);
```

Comment:
I think it's a very nice catch you found this bug. But wouldn't it be a cleaner approach if, instead, we added the transaction handling into `sss_ldb_modify_permissive()` explicitly? This would prevent us from shooting ourselves in the foot in the future.

Please note that a nested transaction in ldb is essentially a no-op:
```
349   start a transaction
 350 */
 351 int ldb_transaction_start(struct ldb_context *ldb)
 352 {
 353 >-------struct ldb_module *next_module;
 354 >-------int status;
 355 
 356 >-------ldb_debug(ldb, LDB_DEBUG_TRACE,
 357 >------->-------  "start ldb transaction (nesting: %d)",
 358 >------->-------  ldb->transaction_active);
 359 
 360 >-------/* explicit transaction active, count nested requests */
 361 >-------if (ldb->transaction_active) {
 362 >------->-------ldb->transaction_active++;
 363 >------->-------return LDB_SUCCESS;
 364 >-------}
```

Reply:
Also, in either way,  I would prefer to have a `bool transaction_active` on the place where the transaction is added, then set it once transaction_start succeeds and have  a `done:` handler in the function that would always cancel the transaction if it's active and the return code is non-zero. This would prevent us from cancelling the transactions in the future if we refactor the code.